openldap/2.5.13/openldap-2.5.13.patch

```diff
@@ -0,0 +1,436 @@
+-	if ( rc == LDAP_SUCCESS )
++	if ( rc == LDAP_SUCCESS ) {
+ 		rc = ldap_X509dn2bv(&der_dn, dn, (LDAPDN_rewrite_func *)func, flags );
++#ifdef HAVE_WOLFSSL
++		/* see comment in tlso_session_peer_dn */
++		free(der_dn.bv_val);
++#endif
++	}
+ 	return rc;
+ }
+ 
+--- ./libraries/libldap/tls_o.c.dist	2022-07-14 12:09:57.000000000 -0500
++++ ./libraries/libldap/tls_o.c	2024-06-29 11:42:47.505895228 -0500
+@@ -18,7 +18,7 @@
+ 
+ #include "portable.h"
+ 
+-#ifdef HAVE_OPENSSL
++#if defined(HAVE_OPENSSL) || defined(HAVE_WOLFSSL)
+ 
+ #include "ldap_config.h"
+ 
+@@ -37,18 +37,23 @@
+ #include "ldap-int.h"
+ #include "ldap-tls.h"
+ 
+-#ifdef HAVE_OPENSSL_SSL_H
++#if defined(HAVE_OPENSSL_SSL_H) || defined(HAVE_WOLFSSL)
++#ifdef HAVE_WOLFSSL
++#include <wolfssl/options.h>
++#endif
+ #include <openssl/ssl.h>
+ #include <openssl/x509v3.h>
+ #include <openssl/err.h>
+ #include <openssl/rand.h>
++#ifndef HAVE_WOLFSSL
+ #include <openssl/safestack.h>
++#endif
+ #include <openssl/bn.h>
+ #include <openssl/rsa.h>
+ #include <openssl/dh.h>
+ #endif
+ 
+-#if OPENSSL_VERSION_NUMBER >= 0x10100000
++#if OPENSSL_VERSION_NUMBER >= 0x10100000 && !defined(HAVE_WOLFSSL)
+ #define ASN1_STRING_data(x)	ASN1_STRING_get0_data(x)
+ #endif
+ 
+@@ -87,7 +92,7 @@ static void tlso_locking_cb( int mode, i
+ 	}
+ }
+ 
+-#if OPENSSL_VERSION_NUMBER >= 0x0909000
++#if OPENSSL_VERSION_NUMBER >= 0x0909000 && !defined(HAVE_WOLFSSL)
+ static void tlso_thread_self( CRYPTO_THREADID *id )
+ {
+ 	CRYPTO_THREADID_set_pointer( id, (void *)ldap_pvt_thread_self() );
+@@ -125,7 +130,7 @@ static void tlso_thr_init( void ) {}
+ #endif
+ #endif /* OpenSSL 1.1 */
+ 
+-#if OPENSSL_VERSION_NUMBER < 0x10100000
++#if OPENSSL_VERSION_NUMBER < 0x10100000 && !defined(HAVE_WOLFSSL)
+ /*
+  * OpenSSL 1.1 API and later makes the BIO method concrete types internal.
+  */
+@@ -162,6 +167,7 @@ BIO_meth_free( BIO_METHOD *meth )
+ 
+ #endif /* OpenSSL 1.1 */
+ 
++#ifndef HAVE_WOLFSSL
+ static STACK_OF(X509_NAME) *
+ tlso_ca_list( char * bundle, char * dir, X509 *cert )
+ {
+@@ -195,6 +201,7 @@ tlso_ca_list( char * bundle, char * dir,
+ 	}
+ 	return ca_list;
+ }
++#endif
+ 
+ /*
+  * Initialize TLS subsystem. Should be called only once.
+@@ -222,8 +229,10 @@ tlso_init( void )
+ 	OPENSSL_init_ssl(0, NULL);
+ #endif
+ 
++#ifndef HAVE_WOLFSSL
+ 	/* FIXME: mod_ssl does this */
+ 	X509V3_add_standard_extensions();
++#endif
+ 
+ 	tlso_bio_method = tlso_bio_setup();
+ 
+@@ -262,7 +271,7 @@ static void
+ tlso_ctx_ref( tls_ctx *ctx )
+ {
+ 	tlso_ctx *c = (tlso_ctx *)ctx;
+-#if OPENSSL_VERSION_NUMBER < 0x10100000
++#if OPENSSL_VERSION_NUMBER < 0x10100000 && !defined(HAVE_WOLFSSL)
+ #define	SSL_CTX_up_ref(ctx)	CRYPTO_add( &(ctx->references), 1, CRYPTO_LOCK_SSL_CTX )
+ #endif
+ 	SSL_CTX_up_ref( c );
+@@ -458,8 +467,62 @@ tlso_ctx_init( struct ldapoptions *lo, s
+ 
+ 		if ( is_server ) {
+ 			STACK_OF(X509_NAME) *calist;
++/*
++ * tlso_ca_list calls SSL_add_dir_cert_subjects_to_stack, which wolfSSL hasn't
```

Comment:
Seams like it would be the same effort to just implement `SSL_add_dir_cert_subjects_to_stack` in wolfSSL. Let's leave it like this for now.